src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,150 @@
+     */
+    public static Command parse(String rawCommand) throws DukeException {
+        // identify command name
+        String[] rawSplit = rawCommand.split(" ", 2);
+        String commandName = rawSplit[0];
+        String argString = null;
+        if (rawSplit.length == 2) {
+            argString = rawSplit[1];
+        }
+
+        Command command = null;
+        int index = -1;
+        String[] args = null;
+
+        // create commands depending on the command name
+        switch (commandName) {
+        case "bye":
+            command = new ByeCommand();
+            break;
+        case "list":
+            command = new ListCommand();
+            break;
+        case "mark":
+            // TODO: throw exception in MarkCommand if mark index out of bounds
+            args = Parser.getArgs(commandName, argString, 1);
+            index = Integer.parseInt(args[0]) - 1;
+            command = new MarkCommand(index);
+            break;
+        case "unmark":
+            // TODO: throw exception in UnmarkCommand if mark index out of bounds
+            args = Parser.getArgs(commandName, argString, 1);
+            index = Integer.parseInt(args[0]) - 1;
+            command = new UnmarkCommand(index);
+            break;
+        case "delete":
+            // TODO: throw exception in UnmarkCommand if mark index out of bounds
+            args = Parser.getArgs(commandName, argString, 1);
+            index = Integer.parseInt(args[0]) - 1;
+            command = new DeleteCommand(index);
+            break;
+        case "find":
+            args = Parser.getArgs(commandName, argString, 1);
+            command = new FindCommand(args[0]);
+            break;
+        case "todo":
+            args = Parser.getArgs(commandName, argString, 1, new String[] {});
+            command = new ToDoCommand(args[0]);
+            break;
+        case "deadline":
+            args = Parser.getArgs(commandName, argString, 2, new String[] { "/by" });
+            command = new DeadlineCommand(args[0], args[1]);
+            break;
+        case "event":
+            args = Parser.getArgs(commandName, argString, 3, new String[] { "/from", "/to" });
+            command = new EventCommand(args[0], args[1], args[2]);
+            break;
+        default:
+            throw new UnknownCommandException(commandName);
+        }
+        return command;
+    }
+
+    /**
+     * Parses an argument string into an array of arguments, splitting by space ("
+     * ").
+     * 
+     * @param commandName The name of the current command
+     * @param argString   The argument string to be parsed
+     * @param numArgs     The number of arguments to be parsed
+     * @return An array of the numArgs arguments.
+     * @throws InvalidArgumentException Thrown when argument string does not tally
+     *                                  with numArgs given.
+     */
+    public static String[] getArgs(String commandName, String argString, int numArgs)
+
```

Comment:
Small detail, this could be an extra newline?

Reply:
good catch, thanks